erica/erica_legacy/pyeric/eric_errors.py

```diff
@@ -264,93 +264,11 @@
-                         "ERIC_CRYPT_E_INVALID_PARAM_ABC", "ERIC_CRYPT_E_MAX_SESSION", "ERIC_CRYPT_E_NO_SERVICE",
-                         "ERIC_CRYPT_E_NO_SIG_ENC_KEY", "ERIC_CRYPT_E_OUT_OF_MEM", "ERIC_CRYPT_E_P11_ENC_KEY",
-                         "ERIC_CRYPT_E_P11_ENGINE_LOADED", "ERIC_CRYPT_E_P11_INIT_FAILED",
-                         "ERIC_CRYPT_E_P11_NO_ENC_CERT", "ERIC_CRYPT_E_P11_NO_SIG_CERT", "ERIC_CRYPT_E_P11_SIG_KEY",
-                         "ERIC_CRYPT_E_P12_CREATE", "ERIC_CRYPT_E_P12_DECODE", "ERIC_CRYPT_E_P12_ENC_KEY",
-                         "ERIC_CRYPT_E_P12_NO_ENC_CERT", "ERIC_CRYPT_E_P12_NO_SIG_CERT", "ERIC_CRYPT_E_P12_READ",
-                         "ERIC_CRYPT_E_P12_SIG_KEY", "ERIC_CRYPT_E_P7_DECODE", "ERIC_CRYPT_E_P7_READ",
-                         "ERIC_CRYPT_E_P7_RECIPIENT", "ERIC_CRYPT_E_PIN_LOCKED", "ERIC_CRYPT_E_SC_ENC_KEY",
-                         "ERIC_CRYPT_E_SC_INIT_FAILED", "ERIC_CRYPT_E_SC_NO_APPLET", "ERIC_CRYPT_E_SC_NO_ENC_CERT",
-                         "ERIC_CRYPT_E_SC_NO_SIG_CERT", "ERIC_CRYPT_E_SC_SESSION", "ERIC_CRYPT_E_SC_SIG_KEY",
-                         "ERIC_CRYPT_E_TOKEN_TYPE_MISMATCH", "ERIC_CRYPT_E_USER_CANCEL",
-                         "ERIC_CRYPT_E_VERIFY_CERT_CHAIN", "ERIC_CRYPT_E_XML_SIG_ADD", "ERIC_CRYPT_E_XML_SIG_SIGN",
-                         "ERIC_CRYPT_EIDKARTE_NICHT_UNTERSTUETZT", "ERIC_CRYPT_ERROR_CREATE_KEY",
-                         "ERIC_CRYPT_NICHT_UNTERSTUETZTES_PSE_FORMAT", "ERIC_CRYPT_PIN_STAERKE_NICHT_AUSREICHEND",
-                         "ERIC_CRYPT_SIGNATUR", "ERIC_CRYPT_ZERTIFIKAT",
-                         "ERIC_CRYPT_ZERTIFIKATSDATEI_EXISTIERT_BEREITS", "ERIC_GLOBAL_ARITHMETIKFEHLER",
-                         "ERIC_GLOBAL_DATEI_NICHT_GEFUNDEN", "ERIC_GLOBAL_DATEIZUGRIFF_VERWEIGERT",
-                         "ERIC_GLOBAL_ERR_DEKODIEREN", "ERIC_GLOBAL_INKOMPATIBLE_VERSIONEN",
-                         "ERIC_GLOBAL_INTERNER_FEHLER", "ERIC_GLOBAL_LOG_EXCEPTION",
-                         "ERIC_GLOBAL_NICHT_GENUEGEND_ARBEITSSPEICHER", "ERIC_GLOBAL_NO_VERSAND_IN_BETA_VERSION",
-                         "ERIC_GLOBAL_UNKNOWN_PARAMETER_ERROR", "ERIC_GLOBAL_UPDATE_NECESSARY",
-                         "ERIC_IO_FALSCHES_VERFAHREN", "ERIC_IO_MASTERDATENSERVICE_NICHT_VERFUEGBAR",
-                         "ERIC_IO_VERSIONSINFORMATIONEN_NICHT_GEFUNDEN", "ERIC_TRANSFER_ERR_OTHER",
-                         "ERIC_TRANSFER_ERR_SEND", "ERIC_TRANSFER_ERR_SEND_INIT"]
+ERIC_ERRORS_WITH_RETRY = ["ERIC_CRYPT_E_SC_SLOT_EMPTY", "ERIC_CRYPT_E_P11_SLOT_EMPTY", "ERIC_TRANSFER_ERR_PROXYCONNECT",
+                          "ERIC_TRANSFER_ERR_CONNECTSERVER", "ERIC_TRANSFER_ERR_NORESPONSE",
+                          "ERIC_TRANSFER_ERR_TIMEOUT",
+                          "ERIC_TRANSFER_COM_ERROR", "ERIC_TRANSFER_ERR_ENDDATENGROESSE",
+                          "ERIC_TRANSFER_ERR_NOTENCRYPTED"]
```

Comment:
I just copied the errors listed in confluence with a useful retry and these two:

- ERIC_TRANSFER_ERR_NOTENCRYPTED
- ERIC_TRANSFER_ERR_ENDDATENGROESSE

are listed as such and these two:

- ERIC_TRANSFER_ERR_SEND
- ERIC_GLOBAL_INTERNER_FEHLER

are not. BUT you have much more knowledge with erica and if you give me another GO! I will add them :)

Reply:
You are right, I thought I added the sentence "But probably a lot of thought already went into this" - definitely should have. But when you read the description of the named errors in the API-Referenz of Eric, would you agree that the two additional ones seem beneficial to retry (and the other two don't)? Because, if you agree, let's exchange them.

Reply:
- ERIC_TRANSFER_ERR_NOTENCRYPTED: [[610101274] Die Antwortdaten waren nicht PKCS#7-verschlüsselt.
- ERIC_TRANSFER_ERR_ENDDATENGROESSE: [610101260] Im XML-String konnte der Text "</DatenGroesse>" nicht
gefunden werden.

The second one would be definitely a non-retryable error but the first one not so sure, maybe something went wrong with the encryption? But sure if something goes wrong is that we couldn't encrypt at all the message and we would get the other SLOT known error, right?

- ERIC_TRANSFER_ERR_SEND: [610101271] Beim Datenaustausch ist ein Fehler aufgetreten.
- ERIC_GLOBAL_INTERNER_FEHLER: [610001032] Interner Fehler aufgetreten. Details stehen ggf. im Logfile
(eric.log).

These two we could add them to the retryable list jep!

So what about:
- We remove ERIC_TRANSFER_ERR_ENDDATENGROESSE from retryable error list.
- We add the other two

Reply:
Great! 💯

Reply:
Done :)